### PR TITLE
Change the long version to space separated + port 0 fixes

### PR DESCRIPTION
**Pull request by ldemailly**
```
$ docker run istio/fortio:webtest version
0.7.2-pre 2018-02-24 14:14 48d91e285fa8d1c6baf7e8bf741ea8a36bbe2fdf
```

also fixes UI URL printout for port 0 (#110)

```
ldemailly-macbookpro:fortio ldemailly$ fortio server --http-port :0
Https redirector running on :8081
Fortio 0.7.2-pre grpc ping server listening on port :8079
Fortio 0.7.2-pre echo server listening on [::]:57063
UI starting - visit:
http://localhost:57063/fortio/
^C
ldemailly-macbookpro:fortio ldemailly$ fortio server --http-port [::1]:0
Https redirector running on :8081
Fortio 0.7.2-pre grpc ping server listening on port :8079
Fortio 0.7.2-pre echo server listening on [::1]:57068
UI starting - visit:
http://[::1]:57068/fortio/
^C
ldemailly-macbookpro:fortio ldemailly$ fortio server --http-port
192.168.1.90:blp5
Https redirector running on :8081
Fortio 0.7.2-pre grpc ping server listening on port :8079
Fortio 0.7.2-pre echo server listening on 192.168.1.90:48129
UI starting - visit:
http://192.168.1.90:48129/fortio/
```

And also: extra test for #83 (all stats bucket boundaries)

### CODE REVIEW

**Comment by ldemailly** [comment]
ps: it's not fixing the report case

**Comment by ldemailly** [comment]
timing out on review- will merge this now; feel free to comment though

File: stats/stats_test.go [hunk 1]
```diff
@@ -765,6 +765,35 @@ func TestBucketLookUp(t *testing.T) {
 	}
 }
 
+func TestAllBucketBoundaries(t *testing.T) {
```
Comment (asadali) [inline stats/stats_test.go:768]: +1

File: ui/uihandler.go [hunk 2]
```diff
@@ -903,7 +899,14 @@ func Serve(baseurl, port, debugpath, uipath, staticRsrcDir string, datadir strin
 		fs := http.FileServer(http.Dir(dataDir))
 		http.Handle(uiPath+"data/", LogAndFilterDataRequest(http.StripPrefix(uiPath+"data", fs)))
 	}
-	fhttp.Serve(hostPort, debugpath)
+	addr := fhttp.Serve(hostPort, debugpath)
+	setHostAndPort(hostPort, addr)
+	uiMsg := fmt.Sprintf("UI starting - visit:\nhttp://%s%s", urlHostPort, uiPath)
+	if !strings.Contains(port, ":") {
```
Comment (asadali) [inline ui/uihandler.go:905]: minor nit: not clear about the use case in which this will be invoked. maybe a comment would have been helpful?

File: ui/uihandler.go [hunk 3]
```diff
@@ -968,11 +972,14 @@ func RedirectToHTTPS(port string) {
 
 // setHostAndPort takes hostport in the form of hostname:port, ip:port or :port,
 // sets the urlHostPort variable and returns hostport unmodified.
-func setHostAndPort(hostport string) string {
-	if strings.HasPrefix(hostport, ":") {
-		urlHostPort = "localhost" + hostport
-		return hostport
+func setHostAndPort(inputPort string, addr *net.TCPAddr) {
```
Comment (asadali) [inline ui/uihandler.go:975]: can add a test for this.

**Review by asadali — COMMENTED** [review]
looks good overall. minor nitpicks.

general question unrelated to PR: how does the submodule update show up in this PR? i.e.:
```
Submodule vendor updated 2 files
```
do I need to execute `make depend.update` to pull in updates to vendor?

**Comment by ldemailly** [comment]
> do I need to execute make depend.update to pull in updates to vendor?

no you just "make pull" or "make submodule-sync"

depend.update is for when you want to change the dependencies

File: ui/uihandler.go (hunk 2 above)
Comment (ldemailly) [reply ui/uihandler.go:905]: yes it's about putting "localhost" when no binding address is requested by the user

File: ui/uihandler.go (hunk 3 above)
Comment (ldemailly) [reply ui/uihandler.go:975]: the whole thing is quite yucky and has no unit tests... so far
